protocol-definitions/ReplicatedMap.yaml

```diff
@@ -852,3 +852,75 @@ methods:
           doc: |
             partition id of the entries
     response: {}
+  - id: 20
+    name: fetchEntryViews
+    since: 2.7
+    doc: |
+      Allows iteration over ReplicatedMapEntryViewHolder objects. The changes happened during the iteration may not be included in the iterated EntryViews. This method
+      will throw an exception if there is no replicated record store with the given replicated map name and partition id. This method will consume some memory in the 
+      member with the default timeout of 300 seconds that is reset after each fetch. Sending endEntryViewIteration or timing out will release the resource.
+    request:
+      retryable: true
+      partitionIdentifier: -1
+      params:
+        - name: name
+          type: String
+          nullable: false
+          since: 2.7
+          doc: |
+            Name of the ReplicatedMap
+        - name: cursorId
+          type: UUID
+          nullable: true
+          since: 2.7
+          doc: |
+            The identifier of the last fetched page. Initially null and then returned by this API. 
+        - name: partitionId
+          type: int
+          nullable: false
+          since: 2.7
+          doc: |
+            The partition ID of the partition that the EntryViews belong to.
+        - name: batchSize
+          type: int
+          nullable: false
+          since: 2.7
+          doc: |
+            The maximum number of EntryViews to be returned.
+    response:
+      params:
+        - name: cursorId
+          type: UUID
+          nullable: false
+          since: 2.7
+          doc: |
+            A unique string which is used to fetch new pages.
+        - name: entryViews
+          type: List_ReplicatedMapEntryViewHolder
+          nullable: true
+          since: 2.7
+          doc: |
+            A list of EntryViews. Null means end of iteration. 
+  - id: 21
+    name: endEntryViewIteration
+    since: 2.7
+    doc: |
+      This method is used to release the resource generated by fetchEntryViews. Send this request after you retrieved the last page and no
```

Comment:
why do we need the client to make this request? Server knows it finished and it should release the resources. What is the rationale to make server cleanup depend on a client request?

Reply:
We need this extra request to tell the server that the client has received the last page of data.

If we don't do this the following could happen:
- client sends a request - it is the last page
- server clears the cursor
- response with the last page gets lost
- client retries, but server doesn't have the data anymore.

However, I don't think we need a new request type for this, we can send another request when client receives batch < max batch size (or empty), the client knows the response to this will be empty, so it doesn't even need to wait for that 2nd response.

Reply:
If page size divides entry count evenly, that logic won't work. I can add another parameter "endIteration:true" to the original method but I prefer a new method because it resembles the SQL close statement.

Reply:
I think by empty you mean null page. it can work but I don't see a problem in a new method